girder/models/item.py

```diff
@@ -430,13 +434,20 @@ def fileList(self, doc, user=None, path='', includeMetadata=False,
                     (includeMetadata and doc.get('meta', {}))):
                 path = os.path.join(path, doc['name'])
         metadataFile = 'girder-item-metadata.json'
+
+        if stream:
+            returnFunc = functools.partial(self.model('file').download,
+                                           headers=False)
+        else:
+            def returnFunc(item):
+                return item
```

Comment:
Negligible difference. Opt for readability / simplicity over premature optimization.